platforms/jvm/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileAvoidanceWithBuildCacheServiceIntegrationTest.groovy

```diff
@@ -62,6 +74,70 @@ class JavaCompileAvoidanceWithBuildCacheServiceIntegrationTest extends AbstractI
         executedAndNotSkipped ':b:jar', ':a:compileJava'
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/32464")
+    def "kotlin library on build script classpath and Java classpath doesn't affect Java compilation build cache key"() {
+        given:
+        // We need to work with remote Kotlin libraries with inline functions to reproduce the cache issue
+        def repo = publishKotlinLibraryWithInlineFunctionToRemote()
+        def kotlinLibraryWithInlineFunction = "com.example:kotlin-library:1.0.0"
+
+        setupBaseProjectA()
```

Comment:
Do you intend to call this fixture setup method it from other tests? Otherwise, I would inline it/merge with the setup code below